rfcs/41-ui-navigation.md

```diff
@@ -0,0 +1,394 @@
+      )
+    )
+  )
+}
+fn react_events(events: ResMut<Event<NavEvent>>, game: Res<Game>) {
+  for event in events.iter() {
+    if matches!(event, NavEvent::Caught(NavCommand::Action, game.ui.start_button)) {
+      // start game
+    }
+    //etc.
+  }
+}
+```
+
+
+## Implementation strategy
+
+### Basic physical navigation
+
+Let's drop the idea to change submenu with LT/RT for a while. We would change
+submenus by navigating upward with the Dpad or arrow keys to the tab bar and
+selecting the tab with the LEFT and RIGHT keys.
+
+Typical game engine implementation of menu navigation requires the developer to
+specify the "next", "left", "right", "previous" etc.[^1] relationships between
+focusable elements. This is just lame! Our game engine not only already knows
+the position of our UI elements, but it also has knowledge of the logical
+organization of elements through the `Parent`/`Children` relationship. (in this
+case, children are specified in the parenthesis following the element name,
+look at `grid(...)`)
+
+Therefore, with the little information we provided through this hand wavy
+specification, we should already have everything setup for the navigation to
+just work™.
+
+
+### Dimensionality
+
+Let's go back to our ambition of using LT/RT for navigation now. 
+
+UI navigation is not just 2D, it's N-dimensional. The
+LT/RT to change tabs is like navigating in a 3rd orthogonal dimension to the
+dimensions you navigate with UP DOWN LEFT RIGHT inside the menus.
+
+I'll call those dimensions `Plane`s because it's easier to type than
+"Dimension", I'll limit the implementation to 3 planes. We can imagine truly
+exotic menu navigation systems with more than 3 dimensions, but I'll not
+worry about that. Our `Plane`s are:
+* `Plane::Menu`: Use LT/RT to move from one element to the next/previous
+* `Plane::Select`: Instead of emitting an `Action` event when left-clicking or
+  pressing A/B, go up-down that direction
+* `Plane::Physical`: Use the `Transform` positioning and navigate with Dpad or
+  arrow keys
+
+Each `NavCommand` moves the focus on a specific `Plane` as follow:
+* `Plane::Menu`: `Previous`, `Next`
+* `Plane::Select`: `Action`, `Cancel`
+* `Plane::Physical`: `MoveUp`, `MoveDown`, `MoveLeft`, `MoveRight`
+
+We should also be able to "loop" the menu, for example going LEFT of "soul"
+loops us back to "all" at the other side of the tab bar.
+
+### Specifying navigation dimensions
+
+We posited we could easily infer the physical layout and bake a navigation map
+automatically based on the `Transform` positions of the elements. This is not
+the case for the dimensions of our navigation. We can't magically infer the
+intent of the developer: They need to specify the dimensionality of their menus.
+
+Let's go back to the menu. In the example code, I refer to `:container` and
+`:navigable`. I've not explained yet what those are. Let's clear things up.
+
+A _navigable_ is an element that can be focused. A _container_ is a node entity
```

Comment:
I suspect we can just reuse the existing `Interaction` component, probably reflavoring to `Focus`. The venn diagram is a circle. so we can use the same component for both marking and storing the state.